.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,22 @@
+##### Motivation for the changes
+
+<!-- An issue reference or explanation why the change is desirable. -->
+<!-- This speeds up step 1 of the process. -->
+
+
+
+
+
+
+##### Checklist for maintainers
+
+<!-- Contributors: please leave this as is -->
+<!-- Maintainers: tick if completed or explain if not relevant -->
+
+ - [ ] is the idea good? has it been discussed by the Nix team?
+ - [ ] unit tests
+ - [ ] functional tests (`tests/**.sh`)
+ - [ ] documentation in the manual
+ - [ ] documentation in the code (if necessary; ideally code is already clear)
```

Comment:
```suggestion
- [ ] code and comments are self-explanatory
```

I was trying to make it more tangible what "documentation in the code" is supposed to mean, but it's really hard to come up with something clear-cut.

I want code to be "readable" in a sense that if you read it, it answers all the questions about itself without having to dig around. That doesn't necessarily mean there need to be lots of comments. The symbol names should allow one to easily infer what they mean, comments should explain why the code is written the way it is, or explain what it does if there is no clean way of expressing it in the code itself. There is plenty of material out there that explains really well why these (and others) are qualities of readable code, and eventually we should collect these resources for reference.